dev/NavigationView/NavigationView_ApiTests/NavigationViewTests.cs

```diff
@@ -783,17 +783,36 @@ public void VerifyExpandCollapseChevronVisibility()
                 UIElement chevronUIElement = (UIElement)VisualTreeUtils.FindVisualChildByName(parentItem, "ExpandCollapseChevron");
                 Verify.IsTrue(chevronUIElement.Visibility == Visibility.Collapsed, "chevron should have been collapsed as NavViewItem has no children");
 
-                // Add a child to our NavigationView parentItem. This should make the chevron visible.
-                children.Add("Undo");
+                // Add a child to parentItem through the MenuItemsSource API. This should make the chevron visible.
+                children.Add("Child 1");
                 Content.UpdateLayout();
 
                 Verify.IsTrue(chevronUIElement.Visibility == Visibility.Visible, "chevron should have been visible as NavViewItem now has children");
 
-                // Remove all children from our NavigationView parentItem. This should collapse the chevron
+                // Remove all children of parentItem. This should collapse the chevron
                 children.Clear();
                 Content.UpdateLayout();
 
                 Verify.IsTrue(chevronUIElement.Visibility == Visibility.Collapsed, "chevron should have been collapsed as NavViewItem no longer has children");
+
+                // Add a child to parentItem and set the MenuItemsSource as null. This should collapse the chevron
+                children.Add("Child 2");
+                parentItem.MenuItemsSource = null;
```

Comment:
Do we need a `Content.UpdateLayout()` call here to make sure the chevron is visible again before setting the MenuItemsSource to null and verifying that the chevron is collapsed? (Before adding "Child 2" item, the chevron is collapsed.)

Reply:
But we have already checked earlier at Line 787 that adding an item to children does make the chevron visible right? I think that it would be redundant to make that check again here?

Reply:
My thinking here is not to **verify** that adding an item actually makes the chevron visible (by adding an additional Verify.IsTrue(chevronUIElement.Visibility == Visibility.Visible)) but making sure that by the time we set `parentItem.MenuItemsSource = null;` the chevron is actually visible. That way, we can make sure that setting MenuItemsSource to null will actually collapse the chevron.

In other words, my concern here is the following: Before we add our child item (Child 2) the chevron is collapsed. We then add this child and immediately after that set MenuItemsSource to null. I'm concerned that without the `Content.UpdateLayout()` call after adding the child, the chevron might not actually be made visible again, thus we are not actually verifiying whether setting MenuItemsSource to null *collapses* the chevron. (Since the chevron might still be collapsed when we set the source to null and thus the `Verify.IsTrue()` call below succeeds, even though `MenuItemsSource  = null` might not actually change the chevron visibility state at all.)

@StephenLPeters @chingucoding Your thoughts here?

Reply:
I understand now. Thanks. Will add those now.

Reply:
Valid concern, and another Verify would guarantee that was the case. Without it it could be the case that a bug causes the chevron to never show which wouldn't be caught without the "redundant" verify.